packages/visibility_detector/lib/src/render_visibility_detector.dart

```diff
@@ -34,33 +98,220 @@ class RenderVisibilityDetector extends RenderProxyBox {
 
   /// Used by [VisibilityDetector.updateRenderObject].
   set onVisibilityChanged(VisibilityChangedCallback? value) {
+    if (_onVisibilityChanged == value) {
+      return;
+    }
+    _compositionCallbackCanceller?.call();
+    _compositionCallbackCanceller = null;
     _onVisibilityChanged = value;
-    markNeedsCompositingBitsUpdate();
-    markNeedsPaint();
+
+    if (value == null) {
+      // Remove all cached data so that we won't fire visibility callbacks when
+      // a timer expires or get stale old information the next time around.
+      forget(key);
+      _lastVisibility.remove(key);
```

Comment:
Wondering, why does forget not also clear the lastVisibility for the key? Is there a use case where you want to forget, but still keep that historical information?